### PR TITLE
Home task for Python Basics module

### PR DESCRIPTION
**Pull request by DenisKrupkin**

### CODE REVIEW

File: Python Basics/python_basics_home_task.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Home task for Module 1 - Python Basics
+# Create list of 100 random numbers from 0 to 1000
```
Comment (arastopchin) [inline Python Basics/python_basics_home_task.py]:
```suggestion
"""
Home task for Module 1 - Python Basics

Create a python script:
create list of 100 random numbers from 0 to 1000
sort list from min to max (without using sort())
calculate average for even and odd numbers
print both average result in console
"""
```
Let's agree to put a task full description to the very beginning on the file like this.

File: Python Basics/python_basics_home_task.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import random  # Import of random module for using of random function
+
+list_of_random = []  # Define the empty list
+for j in range(1000):  # Define of FOR statement with range of iterations from 0 to 999
+    list_of_random.append(random.randint(0, 999))  # Adding of randomly generated by randint function INT values to list
+print('List with generated random values:', list_of_random)  # Printing final list to console
+# Sort list from min to max (without using sort())
```
Comment (arastopchin) [inline Python Basics/python_basics_home_task.py]:
```suggestion
print('List with generated random values:', list_of_random)  # Printing final list to console

# Sort list from min to max (without using sort())
```
I would add some space between logic steps for better readability.

File: Python Basics/python_basics_home_task.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Home task for Module 1 - Python Basics
+# Create list of 100 random numbers from 0 to 1000
+import random  # Import of random module for using of random function
+
+list_of_random = []  # Define the empty list
```
Comment (arastopchin) [inline Python Basics/python_basics_home_task.py]:
```suggestion
random_numbers = []  # Define the empty list
```
It's not clear from your variable name random what? I think random_numbers for instance will show what it does contain.

File: Python Basics/python_basics_home_task.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+for j in range(1000):  # Define of FOR statement with range of iterations from 0 to 999
+    list_of_random.append(random.randint(0, 999))  # Adding of randomly generated by randint function INT values to list
```
Comment (arastopchin) [inline Python Basics/python_basics_home_task.py]: Usually list comprehesions and dict comrehensions are used for creation lists or dicts. I'm not aware if it was in lectures, but let's create this list using list comprehension. It's a best practice in Python.

File: Python Basics/python_basics_home_task.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+print('List with generated random values:', list_of_random)  # Printing final list to console
+# Sort list from min to max (without using sort())
+for run in range(len(list_of_random) - 1):  # Define of initial FOR statement with range of iterations minus 1 from list
+    for i in range(len(list_of_random) - 1 - run):  # Inner FOR statement with decreasing iterations for each run
+        if list_of_random[i] > list_of_random[i + 1]:  # Compare two adjacent values from list
+            list_of_random[i], list_of_random[i + 1] = list_of_random[i + 1], list_of_random[i]  # Swap places
```
Comment (arastopchin) [inline Python Basics/python_basics_home_task.py]:
```suggestion
for run in range(len(list_of_random)-1):  # Define of initial FOR statement with range of iterations minus 1 from list
    for i in range(len(list_of_random)-1-run):  # Inner FOR statement with decreasing iterations for each run
        if list_of_random[i] > list_of_random[i+1]:  # Compare two adjacent values from list
            list_of_random[i], list_of_random[i+1] = list_of_random[i+1], list_of_random[i]  # Swap places
```
PEP8

File: Python Basics/python_basics_home_task.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+print('Sorted from min to max list:', list_of_random)  # Printing of sorted by bubble sorting list
+# Calculate average for even and odd numbers
+even_numbers = []  # Define empty list for even numbers
+odd_numbers = []  # Define empty list for odd numbers
+for i in range(len(list_of_random)):  # Define FOR statement with number of iterations equal to list length
+    if list_of_random[i] % 2 == 0:  # Check if number can be divided by 2 without reminder
+        even_numbers.append(list_of_random[i])  # Adding of even values to separate list
+    else:
+        odd_numbers.append(list_of_random[i])  # Else other values added to odd values list
```
Comment (arastopchin) [inline Python Basics/python_basics_home_task.py]: 2 things here:
1) If you want to iterate through iterable collection, you do not need to use its indeses each time. Simple 'for' loop can also take element itself. For example:
```suggestion
for number in list_of_random:
    if number % 2 == 0:  # Check if number can be divided by 2 without reminder
        even_numbers.append(number)  # Adding of even values to separate list
    else:
        odd_numbers.append(number)  # Else other values added to odd values list
```
2) There are another options how to do it, it's not required to change current implementation, but just FYI: you can use list comprehension or 'filter' function as well. If you're interested you can find such examples in the Internet.

File: Python Basics/python_basics_home_task.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    even_average = total / len(even_numbers)  # Calculation of average even number
+except ZeroDivisionError:  # In case of zero division error
+    print('Zero division error.')  # Printing message
+for i in range(len(odd_numbers)):  # Define of FOR statement with range equal to odd numbers list length
+    total = 0  # Clearing of total with 0 value
+    total = total + odd_numbers[i]  # Calculation of total sum
+try:  # Zero division error handling
+    odd_average = total / len(odd_numbers)  # Calculation of average odd number
+except ZeroDivisionError:  # In case of zero division error
+    print('Zero division error.')  # Printing message
```
Comment (arastopchin) [inline Python Basics/python_basics_home_task.py]: I would just replace it with 'avg' built-in function (math module? don't remember) and then you wouldn't need to catch ZeroDivision exception. But it's nonetheless a good practice.

File: Python Basics/python_basics_home_task.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+print('Average count of even numbers:', even_average)  # Printing of average of even numbers to console
+print('Average count of odd numbers:', odd_average)  # Printing of average of odd numbers to console
```
Comment (arastopchin) [inline Python Basics/python_basics_home_task.py]: You can try here f-strings, it's very good option:
```suggestion
print(f'Average count of even numbers: {even_average}.')  # Printing of average of even numbers to console
print(f'Average count of odd numbers: {odd_average}.')  # Printing of average of odd numbers to console
```

**Review by arastopchin — CHANGES_REQUESTED** [review]

File: Python Basics/python_basics_home_task.py (hunk 7 above)
Comment (DenisKrupkin) [reply Python Basics/python_basics_home_task.py]: Replaced with mean() function from statistics module.

File: Python Basics/python_basics_home_task.py (hunk 4 above)
Comment (DenisKrupkin) [reply Python Basics/python_basics_home_task.py]: Lists recreated using list compherentions.

**Review by arastopchin — APPROVED** [review]
Brilliant!

**Pull request merged**